.github/workflows/build-test-publish-tbd-artifactory.yml

```diff
@@ -0,0 +1,177 @@
+# Runs on every commit to main. This is the main CI job; it runs in MacOS and Ubuntu environments which:
+# * Build
+# * Run tests
+#
+# In the Ubuntu environment only, to avoid double uploads from MacOS, it also:
+# * Uploads Test reports to BuildKite
+# * Uploads Coverage reports to CodeCov
+# * Uploads Web5 Test Vectors reports to the SDK Report Runner
+# * Publishes (deploys) to TBD's Artifactory instance
+#
+# If triggered from workflow_dispatch, you may select a branch or tag to
+# deploy as an internal "release" (or SNAPSHOT, depending upon the version in the POM)
+# to TBD's Artifactory instance by not specifying a version.
+name: Build, Test, and Publish to TBD Artifactory
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to publish. For example "1.0.0-SNAPSHOT". If not supplied, will default to version specified in the POM. Must end in "-SNAPSHOT".'
+        required: false
+        default: '0.0.0-SNAPSHOT'
+  push:
+    branches:
+      - main
```

Comment:
You removed the trigger on every PR branch. This way our only CI point won't run for PRs. Here we should add back the pull request trigger and skip the snapshot publication to artifactory... (or publish to artifactory anyway since its a snapshot)

Reply:
Oh great catch @leordev would you mind adding those back? Sure, publish to artifactory SNAPSHOTs anyway because then folks could consume PRs to test, could be helpful.

Reply:
Done!